tpp-app/src/services/LogSymptomsService.js

```diff
@@ -52,21 +52,66 @@ export const POSTsymptomsForDate = async (day, month, year, symptoms) => new Pro
  * date.month is a number (January = 1),
  * date.year is a number.
  */
-export const LogMultipleDayPeriod = async (dates) => {
+export const LogMultipleDayPeriod = async (dates) => new Promise(async (resolve, reject) => {
     // run this code for each value in the dates array
-    dates.map(async (date)=>{
-        const year = date.year;
-        const month = date.month;
-        const day = date.day;
+    if(dates.length > 0){
+        try {
+            curYear = dates[0].year;
+            const calendarData = await getCalendarByYear(curYear);
+        
 
-        let symptoms = await GETsymptomsForDate(day, month, year)
+            dates.map((date)=>{
+                const year = date.year;
+                const month = date.month;
+                const day = date.day;
+                try {
+                    
+                    let symptoms = getSymptomsFromCalendar(calendarData, day, month, year);
 
-        if (symptoms.flow == null || symptoms.flow == FLOW_LEVEL.NONE){
-            symptoms.flow = FLOW_LEVEL.MEDIUM;
-        }
+                    if (symptoms.flow == null || symptoms.flow == FLOW_LEVEL.NONE){
+                        symptoms.flow = FLOW_LEVEL.MEDIUM;
+                    }
+
+                    calendarData[year][month-1][day-1] = symptoms;
+                } catch (error) {
+                    console.log(error);
+                }
+
+            })
+
+            if(calendarData[curYear]){
 
-        POSTsymptomsForDate(day, month, year, symptoms);
+                await AsyncStorage.setItem(curYear.toString(), JSON.stringify(calendarData[curYear]))
```

Comment:
doesn't really make sense to use both `await` and `.then` syntax. Imo just use '.then'

Reply:
True!